src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -231,6 +255,7 @@ private function processMessage(Database $dbForProject, Document $message): void
         $message->setAttribute('deliveryErrors', $deliveryErrors);
 
         if (\count($message->getAttribute('deliveryErrors')) > 0) {
+            // TODO: Does this make sense? Only some of the messages might have failed, but we mark the whole message as failed.
```

Comment:
I think this is fine, especially if we clearly document that "failed" means there were some failures.